src/cbmc_starter_kit/update.py

```diff
@@ -0,0 +1,184 @@
+          use these tests."""},
+        {'flag': '--no-update',
+         'action': 'store_true',
+         'help': """
+          Do not update Makefile.common and run-cbmc-proofs.py under CBMC_ROOT/proofs.
+          Normally update these files with copies in this version of the current starter kit."""},
+        {'flag': '--no-starter-kit-removal',
+         'action': 'store_true',
+         'help': """
+          Do not try to remove the CBMC starter kit submodule.
+          Normally check for existence of submodule but remove only with --force."""},
+        {'flag': '--force',
+         'action': 'store_true',
+         'help': """
+          Remove the CBMC starter kit submodule if it exists.
+          This positive confirmation is a safety check that is required before removing
+          any repository submodules."""},
+    ]
+    args = arguments.create_parser(options, desc).parse_args()
+    arguments.configure_logging(args)
+    return args
+
+################################################################
+
+def validate_cbmc_root(args):
+    args.cbmc_root = args.cbmc_root or Path('.')
+    args.cbmc_root = args.cbmc_root.resolve()
+    if not args.cbmc_root.is_dir():
+        raise UserWarning(f'CBMC root is not a directory: {args.cbmc_root}')
+    if not (args.cbmc_root/util.PROOF_DIR).is_dir():
+        raise UserWarning(f'CBMC root is missing a {util.PROOF_DIR} subdirectory: {args.cbmc_root}')
+    logging.debug('CBMC root: %s', args.cbmc_root)
+    return args
+
+def validate_starter_kit_root(args):
+    starter_kit = repository.starter_kit_root(repo=args.cbmc_root, abspath=True)
+    args.starter_kit_root = args.starter_kit_root or starter_kit
+    if args.starter_kit_root:
+        args.starter_kit_root = args.starter_kit_root.resolve()
+        if not args.starter_kit_root.is_dir():
+            raise UserWarning(f'Starter kit root is not a directory: {args.starter_kit_root}')
+        if not (args.starter_kit_root/util.REPOSITORY_TEMPLATES).is_dir():
+            raise UserWarning(f'Starter kit root is missing a {util.REPOSITORY_TEMPLATES} '
+                              f'subdirectory: {args.starter_kit_root}')
+        if not args.starter_kit_root.is_relative_to(args.cbmc_root):
+            raise UserWarning(f'Starter kit root is {args.starter_kit_root} is not a descendant of '
+                              f'CBMC root {args.cbmc_root}')
+    else:
+        args.no_migrate = True
+    logging.debug('CBMC starter kit root: %s', args.starter_kit_root)
+    return args
+
+################################################################
+
+def files_under_root(root=None, symlinks_only=False):
+    cmd = ['find', '.']
+    if symlinks_only:
+        cmd += ['-type', 'l']
+    kwds = {
+        'cwd': root,
+        'text': True,
+        'capture_output': True,
+    }
+    files = subprocess.run(cmd, **kwds, check=True).stdout.splitlines()
```

Comment:
Should this be `check=False` unless you want the CalledProcessError to bubble up to a user?

Reply:
No, if this fails, something is so badly broken it should be an emergency.  Alternatively, I could log.critical and sys.exit(1).